runner/runner.go

```diff
@@ -171,6 +171,8 @@ func (r *Runner) Run(ctx context.Context, query ...string) error {
 			Query: q,
 			Limit: r.options.Limit,
 		}
+		counterLimit := 0
```

Comment:
- This will limit only the output for a specific query
- `counterLimit` is accessed in R/W mode by multiple go routines (in case of multiple agents), this creates a potential data race condition, it would be better to always use atomics in similar scenarios